server/app/auth/oidc/OidcClientProviderParams.java

```diff
@@ -0,0 +1,35 @@
+package auth.oidc;
+
+import auth.ProfileFactory;
+import com.google.auto.value.AutoValue;
+import com.google.errorprone.annotations.RestrictedApi;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import javax.inject.Provider;
+import repository.AccountRepository;
+
+/** Class that holds parameters required by OidcClientProvider and its subclasses. */
+@AutoValue
+public abstract class OidcClientProviderParams {
+  public static OidcClientProviderParams create(
+      Config configuration,
+      ProfileFactory profileFactory,
+      Provider<AccountRepository> accountRepositoryProvider) {
+    return new AutoValue_OidcClientProviderParams(
+        configuration, profileFactory, accountRepositoryProvider);
+  }
+
+  // Our tests have paths like:
+  //   /usr/src/server/test/auth/ProfileMergeTest.java
+  @RestrictedApi(explanation = "Only allow for tests", allowedOnPath = ".*/test/.*")
```

Comment:
**praise:** nice, i did not know about this java feature!

Reply:
I found out about it from the [Javadoc](https://javadoc.io/doc/com.google.guava/guava/31.0.1-jre/com/google/common/annotations/VisibleForTesting.html) for `@VisibleForTesting`, which is apparently advisory-only!